mesonbuild/environment.py

```diff
@@ -225,6 +288,45 @@ def merge_options(self, options):
                 if type(oldval) != type(value):
                     self.coredata.user_options[name] = value
 
+    @staticmethod
+    def get_gnu_compiler_defines(compiler):
+        """
+        Detect GNU compiler platform type (Apple, MinGW, Unix)
+        """
+        # Arguments to output compiler pre-processor defines to stdout
+        # gcc, g++, and gfortran all support these arguments
+        args = [compiler, '-E', '-dM', '-']
+        p = subprocess.Popen(args, universal_newlines=True,
+                             stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+        output = p.communicate('')[0]
+        if p.returncode != 0:
+            raise EnvironmentException('Unable to detect GNU compiler type:\n' + output)
+        try:
+            # Parse several lines of the type:
+            # `#define ___SOME_DEF some_value`
+            # and extract `___SOME_DEF`
+            defines = {}
+            for line in output.split('\n'):
+                if not line:
+                    continue
+                d, key, val = line.split(' ', 2)
+                if d != '#define':
+                    continue
+                defines[key] = val
+        except:
```

Comment:
Gone! Now we don't have any try...except. The code is sufficiently robust in my opinion.